Source/EmbeddedBoundary/ParticleScraper.H

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2021 Andrew Myers
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef PARTICLESCRAPER_H_
+#define PARTICLESCRAPER_H_
+
+#include <AMReX.H>
+#include <AMReX_Vector.H>
+#include <AMReX_MultiFab.H>
+
+#include <EmbeddedBoundary/LevelSet.H>
+
+template <class PC, class F, std::enable_if_t<amrex::IsParticleContainer<PC>::value, int> foo = 0>
+void
+scrapeParticles (PC& pc, const amrex::Vector<const amrex::MultiFab*>& level_set, int lev, F&& f)
+{
+    scrapeParticles(pc, level_set, lev, lev, std::forward<F>(f));
+}
+
+template <class PC, class F, std::enable_if_t<amrex::IsParticleContainer<PC>::value, int> foo = 0>
+void
+scrapeParticles (PC& pc, const amrex::Vector<const amrex::MultiFab*>& level_set, F&& f)
+{
+    scrapeParticles(pc, level_set, 0, pc.finestLevel(), std::forward<F>(f));
+}
+
+template <class PC, class F, std::enable_if_t<amrex::IsParticleContainer<PC>::value, int> foo = 0>
+void
+scrapeParticles (PC& pc, const amrex::Vector<const amrex::MultiFab*>& level_set,
+                 int lev_min, int lev_max, F&& f)
+{
+    BL_PROFILE("scrapeParticles");
+
+    for (int lev = lev_min; lev <= lev_max; ++lev)
+    {
+        const auto dxi = pc.Geom(lev).InvCellSizeArray();
+        for(WarpXParIter pti(pc, lev); pti.isValid(); ++pti)
+        {
+            const auto getPosition = GetParticlePosition(pti);
+            auto& tile = pti.GetParticleTile();
+            auto ptd = tile.getParticleTileData();
+            const auto np = tile.numParticles();
+            auto phi = (*level_set[lev])[pti].array();  // signed distance function
```

Comment:
Is there an issue here if `lev` does not correspond to `maxLevel()`?
I am asking because of this line: 
https://github.com/ECP-WarpX/WarpX/pull/2069/files#diff-1450fa5afa66d2e4dde0bd48fe21eb084d339554ad937d3604376713e053e3c4R1484
which seems to only define the `level_set` for `maxLevel`...

Reply:
There would indeed be an issue here - I was following what was done for the other EB-specific multifabs, which don't work with MR at the moment. However, these functions were written in a way such that they should work once this restriction is lifted. To make this a little safer, I have added an assertion on maxLevel() in case that anyone tries to use this with MR for now.

Reply:
OK, thanks!